docs/reference_guide.md

```diff
@@ -99,7 +99,7 @@ This instruments the tcp_v4_connect() kernel function using a kprobe, with the f
 - ```struct pt_regs *ctx```: Registers and BPF context.
 - ```struct sock *sk```: First argument to tcp_v4_connect().
 
-The first argument is always ```struct pt_regs *```, the remainder are the arguments to the function (they don't need to be specified, if you don't intend to use them).
+The first argument is always ```struct pt_regs *```, the remainder are the arguments to the function (they don't need to be specified, if you don't intend to use them). Limitations: maximum of 4 args including ctx.
```

Comment:
Sorry, I ran into what I thought was a limitation on the number of args
being passed to BPF when in fact it was a bpf_trace_printk throwing the
error. Is there any reason we shouldn't check for the number of args passed
and throw a more helpful error? I wouldn't mind trying to implement that if
no one else opposed. Sorry if that last question is naive but I have never
worked with clang/llvm before, but it looks like it should be possible to
perform a check like in BTypeVisitor::VisitFunctionDecl()


Cryptic error thrown when passing more than 3 args to bpf_trace_printk()

error: <unknown>:0:0: in function kprobe__tcp_rcv_established i32
(%struct.pt_regs*): too many args to 0x7fa3ae20de90: i64 = Constant<6>

On Mon, Aug 22, 2016 at 8:51 AM, Brenden Blanco <notifications@github.com>
wrote:

> In docs/reference_guide.md
> <https://github.com/iovisor/bcc/pull/672#discussion_r75703792>:
>
> > @@ -99,7 +99,7 @@ This instruments the tcp_v4_connect() kernel function using a kprobe, with the f
> >  - ```struct pt_regs *ctx```: Registers and BPF context.
> >  - ```struct sock *sk```: First argument to tcp_v4_connect().
> >
> > -The first argument is always ```struct pt_regs *```, the remainder are the arguments to the function (they don't need to be specified, if you don't intend to use them).
> > +The first argument is always ```struct pt_regs *```, the remainder are the arguments to the function (they don't need to be specified, if you don't intend to use them). Limitations: maximum of 4 args including ctx.
>
> It should be higher, 6 + ctx is what the rewriter supports. The test for
> overflow is in https://github.com/iovisor/bcc/blob/master/tests/python/
> test_clang.py#L345
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/iovisor/bcc/pull/672/files/c9e4e730d3f8f70036e3431c173e0d0d37c971bc#r75703792>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAasKpXHGDUFjpRpJuUAMSvg4bgTViwwks5qicWWgaJpZM4JpcpF>
> .
>